lsp/source/server.mts

```diff
@@ -310,11 +309,23 @@ connection.onDefinition(async ({ textDocument, position }) => {
   assert(program)
 
   return definitions.map<Location | undefined>((definition) => {
-    const { fileName, textSpan } = definition
+    let { fileName, textSpan } = definition
     // TODO: May need to remap fileNames back to sourceFileNames
     const sourceFile = program.getSourceFile(fileName)
     if (!sourceFile) return
 
+    // Reverse map back to .civet source space if we forward mapped earlier
+    if (meta) {
+      const { sourcemapLines } = meta
+      return {
+        uri: service.getSourceFileName(fileName),
+        range: {
+          start: remapPosition(sourceFile.getLineAndCharacterOfPosition(textSpan.start), sourcemapLines),
+          end: remapPosition(sourceFile.getLineAndCharacterOfPosition(textSpan.start + textSpan.length), sourcemapLines),
+        }
+      }
+    }
```

Comment:
This is a fair amount of duplicated code with the blow. Perhaps worth something like:

```js
let start = sourceFile.getLineAndCharacterOfPosition(textSpan.start)
let end = sourceFile.getLineAndCharacterOfPosition(textSpan.start + textSpan.length)
if (meta) {
  const { sourcemapLines } = meta
  start = remapPosition(start, sourcemapLines)
  end = remapPosition(end, sourcemapLines)
}
return {
  uri: service.getSourceFileName(fileName),
  range: {start, end}
}